src/java/picard/sam/FilterSamReads.java

```diff
@@ -181,24 +195,33 @@ protected int doWork() {
             IOUtil.assertFileIsReadable(INPUT);
             IOUtil.assertFileIsWritable(OUTPUT);
             if (WRITE_READS_FILES) writeReadsFile(INPUT);
-
+            final SamReader samReader = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT);
+            
             switch (FILTER) {
                 case includeAligned:
-                    filterReads(new FilteringIterator(SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(INPUT).iterator(),
+                    filterReads(new FilteringIterator(samReader.iterator(),
```

Comment:
+1 to extracting that long repeated line into a variable, but why not go all the way and move the iterator creation to the same place?